MCT/server-files/telemetry.js

```
@@ -9,20 +9,33 @@ Then it adds it to the hisorical and realtime servers through the history and li
  */
 const variables = require('./state-variables.js')
 const request = require('request');
+var path = require('path');
 
 /**
  * The URL of the Elastic Search database
  */
 var searchURl = 'http://localhost:5000/search-es';
-/**
- * The index of the Elastic Search database
- */
-var searchIndex = 'statefield_report_123456789';
 
 /**
  *  constructor initializing and then calling for the generation and updating of all telemetry points from all subsytems and domains
  */
-function Telemetry() {
+function Telemetry(configuration) {
+
+    var FlightSoftware = path.resolve(__dirname, '../..')
+    var config_file = FlightSoftware + "/" + configuration
+    var config_json = require(config_file);
+    /**
+    * The index of the Elastic Search database
+    */
+    if(config_json.devices.length>0){
+      this.searchIndex = 'statefield_report_' + config_json.devices[0].imei
+    }
+    else if(config_json.radios.length>0){
+      this.searchIndex = 'statefield_report_' + config_json.radios[0].imei
+    }
+    else{
+      this.searchIndex = 'statefield_report_'
+    }
```

this would just break everything right? since no imei postfix is provided?

If this branch should never be nominally taken, we should throw an error instead

---

Yea that is fair - To be honest I initially thought that if the config file was missing devices and radios we could just have it defualt to what ci's index is (statefield_report_) but your right, error would be more robust here

---

well if this statefield_report would actually work with ci, then I guess this is okay? I just wanted to check

---

Yea statefield_report_ does work as per ci.json but u might be right about throwing error being better anyways?

---

I changed it to throw an error if there are no devices or radios in the config file.